myia/opt/lib.py

```diff
@@ -819,6 +820,28 @@ def caller_is_marked(g, node, args):
 )
 
 
+@pattern_replacer("just", X, interest=None)
+def expand_composite(resources, node, equiv):
+    """Expand a composite primitive."""
+    if node.is_apply():
+        operation = node.inputs[0]
+        parameters = node.inputs[1:]
+        if operation.is_constant() and isinstance(
+            operation.value, CompositePrimitive
+        ):
```

Comment:
The `is_constant` method takes a type argument precisely for this use case.

```suggestion
        if operation.is_constant(CompositePrimitive):
```